reconcile/slack_usergroups.py

```diff
@@ -689,24 +663,40 @@ def act(
                 usergroup, State()
             )
 
+            slack_client = slack_map[workspace].slack
+
+            desired_ug_state.users = {
```

Comment:
change desired state in `act` is a little strange, since `desired_ug_state.users` only used in `_update_usergroup_users_from_state`, can we construct `users` inside it instead of update desired state?